From 3af58e59d32d6770518aee0df40d509624964662 Mon Sep 17 00:00:00 2001
From: Jesse Keating <jkeating@redhat.com>
Date: Tue, 15 May 2012 13:42:47 -0700
Subject: [PATCH] Use the right argument for dasd module options

The legacy argument is DASD=, and the new arguments should be the same
as the arguments for populating dasd.conf.  If multiple arguments are
passed we can stack them by inserting a ',' between them.
---
 modules.d/95dasd_mod/parse-dasd-mod.sh |   18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

diff --git a/modules.d/95dasd_mod/parse-dasd-mod.sh b/modules.d/95dasd_mod/parse-dasd-mod.sh
index b550156..d7e8664 100755
--- a/modules.d/95dasd_mod/parse-dasd-mod.sh
+++ b/modules.d/95dasd_mod/parse-dasd-mod.sh
@@ -1,12 +1,20 @@
 #!/bin/sh
 # -*- mode: shell-script; indent-tabs-mode: nil; sh-basic-offset: 4; -*-
 # ex: ts=8 sw=4 sts=4 et filetype=sh
-[ -d /etc/modprobe.d ] || mkdir -m 0755 -p /etc/modprobe.d
+mod_args=""
+for dasd_arg in $(getargs rd.dasd= rd_DASD= DASD=); do
+    if [ -z $mod_args ]; then
+        mod_args="$dasd_arg"
+    else
+        # We've already got some thing in mod_args, add to it
+        mod_args="$mod_args,$dasd_arg"
+    fi
+done
 
-dasd_arg=$(getarg rd.dasd_mod.dasd rd_DASD_MOD=)
-if [ -n "$dasd_arg" ]; then
-    echo "options dasd_mod dasd=$dasd_arg" >> /etc/modprobe.d/dasd_mod.conf
+if [ ! -z $mod_args ]; then
+    [ -d /etc/modprobe.d ] || mkdir -m 0755 -p /etc/modprobe.d
+    echo "options dasd_mod dasd=$mod_args" >> /etc/modprobe.d/dasd_mod.conf
 fi
-unset dasd_arg
 
+unset dasd_arg
 dasd_cio_free
